File: C#/6kyu - B.cs
//Link -> https://www.codewars.com/kata/55466989aeecab5aac00003e

using System;
using System.Linq;
using System.Collections.Generic;


public class SqInRect {
  
  public static List<int> sqInRect(int lng, int wdth) {

    List<int> result = new List<int> { };
    if(lng == wdth)
    {
        return null;
    }
    else
    {
        do
        {
            int min = Math.Min(lng, wdth);
            int max = Math.Max(lng, wdth);

            if(lng == max)
            {
               lng = lng - min;
            }
            else
            {
                wdth = wdth - min;
            }     

            result.Add(min);

            } while (Math.Abs(lng - wdth) > 0);

            result.Add(Math.Min(lng, wdth));

            return result;
    }
  }
}

File: scala/6kyu - B.scala
// Link -> https://www.codewars.com/kata/51e0007c1f9378fa810002a9

object Hello extends App{

  def parse(data: String): List[Int] = {
    var value: Int = 0
    var result: List[Int] = List[Int]()

    for(x <- data){
      if (x == 'i'){
        value = value + 1
      }
      else if(x == 'd'){
        value = value - 1
      }
      else if(x == 's'){
        value = value * value
      }
      else if(x == 'o'){
        result = result :+ value
      }
    }

    return result
  }
}
